### PR TITLE
Log and change license only if not equal to the old license

### PR DESCRIPTION
**Pull request by ramizdundar**
EE PR: https://github.com/hazelcast/hazelcast-enterprise/pull/4660

Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Label `Add to Release Notes` or `Not Release Notes content` set
- [x] Request reviewers if possible

### CODE REVIEW

**Review by vbekiaris — APPROVED** [review]

File: hazelcast/src/main/java/com/hazelcast/internal/config/LicenseKey.java [hunk 1]
```diff
@@ -47,6 +47,19 @@ public static String maskLicense(String licenseKey) {
         }
     }
 
+    public static String licenseChangeMessage(boolean persistenceEnabled) {
+        String message = "License updated at run time. ";
+        if (persistenceEnabled) {
+            message += "Dynamic Configuration Persistence is enabled. License will be persisted. "
+                    + "Please create new members with the updated license.";
```
Comment (frant-hartm) [inline hazelcast/src/main/java/com/hazelcast/internal/config/LicenseKey.java:54]:
```suggestion
                    + "Please ensure new members use the updated license.";
```

Comment (frant-hartm) [reply hazelcast/src/main/java/com/hazelcast/internal/config/LicenseKey.java:54]: It is a bit ambiguous - it instructs the user to create new members.

**Review by frant-hartm — CHANGES_REQUESTED** [review]
I would move the message to EE, It's only used from there, reduces coupling between OS and EE repo.

**Comment by ramizdundar** [comment]
> I would move the message to EE, It's only used from there, reduces coupling between OS and EE repo.

In respect to this message I moved the message to EE. I am closing this now.

**Pull request closed without merging**